src/trigrams.rs

```diff
@@ -1,6 +1,6 @@
 use std::collections::HashMap;
 
-pub fn get_trigrams_with_positions(text : &String) -> HashMap<String, u32> {
+pub fn get_trigrams_with_positions(text : &str) -> HashMap<String, u32> {
```

Comment:
I am surprised that Rust does not complain about this type change. I thought it always treats `str` and `String` as different things.